typings/vscode-mssql.d.ts

```diff
@@ -292,6 +305,136 @@ declare module 'vscode-mssql' {
 		validateStreamingJob(packageFilePath: string, createStreamingJobTsql: string): Thenable<ValidateStreamingJobResult>;
 	}
 
+	/**
+	 * Represents a tenant information for an account.
+	 */
+	export interface Tenant {
+		id: string;
+		displayName: string;
+		userId?: string;
+		tenantCategory?: string;
+	}
+
+	/**
+	 * Error that connect method throws if connection fails because of a fire wall rule error.
+	 */
+	export interface IFireWallRuleError extends Error {
+		connectionUri: string;
+	}
+
+	/**
+	 * Represents a key that identifies an account.
+	 */
+	export interface IAccountKey {
+		/**
+		 * Identifier for the account, unique to the provider
+		 */
+		id: string;
+		/**
+		 * Identifier of the provider
+		 */
+		providerId: string;
+		/**
+		 * Version of the account
+		 */
+		accountVersion?: any;
+	}
+
+
+	export enum AccountType {
+		Microsoft = 'microsoft',
+		WorkSchool = 'work_school'
+	}
+
+	/**
+	 * Represents display information for an account.
+	 */
+	export interface IAccountDisplayInfo {
+		/**
+		 * account provider (eg, Work/School vs Microsoft Account)
+		 */
+		accountType: AccountType;
+		/**
+		 * User id that identifies the account, such as "user@contoso.com".
+		 */
+		userId: string;
+		/**
+		 * A display name that identifies the account, such as "User Name".
+		 */
+		displayName: string;
+		/**
+		 * email for AAD
+		 */
+		email?: string;
+		/**
+		 * name of account
+		 */
+		name: string;
+	}
+
+	export interface IAccount {
+		/**
+		 * The key that identifies the account
+		 */
+		key: IAccountKey;
+		/**
+		 * Display information for the account
+		 */
+		displayInfo: IAccountDisplayInfo;
+		/**
+		 * Custom properties stored with the account
+		 */
+		properties: any;
```

Comment:
Can this be given a better type - even just an object with string key/value pairs?

Reply:
I'm worried it breaks some scenarios for AAD.